sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Metrics/ExecutableLinesWalker.cs

```diff
@@ -81,11 +97,31 @@ private void FindExecutableLines(SyntaxNode node)
 
                 case SyntaxKind.ArrayInitializerExpression:
                     executableLineNumbers.Add(node.GetLocation().GetLineNumberToReport());
-                    return;
+                    return true;
 
                 default:
-                    return;
+                    return true;
             }
         }
+
+        private static bool HasExcludedCodeAttribute(SyntaxNode node)
+        {
+            return node
+                .DescendantNodesAndSelf()

```

Comment:
This looks wrong to me.... Why take descendants? If only one method is excluded from code coverage you shouldn't exclude everything.